botocore/awsrequest.py

```diff
@@ -263,6 +263,7 @@ def __init__(self, *args, **kwargs):
             for key, value in self.headers.items():
                 headers[key] = value
         self.headers = headers
+        self.context = {}
```

Comment:
It would be nice to add some documentation explaining what this new public attribute is and what gets set in it.